src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php

```diff
@@ -1151,15 +1151,22 @@ public function originalIsEquivalent($key)
 
         if ($attribute === $original) {
             return true;
-        } elseif (is_null($attribute)) {
-            return false;
-        } elseif ($this->isDateAttribute($key)) {
-            return $this->fromDateTime($attribute) ===
-                   $this->fromDateTime($original);
         }
 
+        // If the attribute is a date, compare it by datetime format
+        if ($this->isDateAttribute($key)) {
+            return $this->fromDateTime($attribute) === $this->fromDateTime($original);
+        }
+
+        // If the original or attribute is an object, compare it non-strict
+        if (is_object($attribute)
+            || is_object($original)) {
+            return $attribute == $original;
+        }
```

Comment:
So, except removing the is_null early return: this is the only addition here. Even the date stuff is not really new, just reformated.

Am I right?

Reply:
Yes at this point true, but it will at least fix the bug as described in the first part of the PR.

Secondly, what do you think about the NOTE part of the PR, i think it should also be ensured that `1` and 1 are considered not equil (may in another PR).